sfopenboson/tests/test_bose_hubbard_prop.py

```diff
@@ -12,187 +12,209 @@
 
 import strawberryfields as sf
 from strawberryfields.ops import Fock, Sgate
 
 from sfopenboson.ops import BoseHubbardPropagation
-from sfopenboson.tests import BaseTest
 
 
-class TestBoseHubbardPropagation(BaseTest):
+class TestBoseHubbardPropagation:
     """Tests for the BoseHubbardPropagation operation"""
-    def setUp(self):
-        """Parameters"""
-        self.hbar = 2.
-        self.eng, _ = sf.Engine(2, hbar=self.hbar)
-        self.J = -1
-        self.U = 1.5
-        self.V = 1/np.sqrt(2)
-        self.t = 1.086
-        self.k = 20
-        self.tol = 1e-2
-
-    def test_1x2_no_onsite(self):
+    J = -1
+    U = 1.5
+    V = 1/np.sqrt(2)
+    t = 1.086
+    k = 20
+
+    @pytest.fixture
+    def eng(self, hbar):
+        """Engine to use for the tests"""
+        sf.hbar = hbar
+        return sf.Engine("fock", backend_options={"cutoff_dim": 7})
+
+    def test_1x2_no_onsite(self, eng, tol):
         """Test a 1x2 lattice Bose-Hubbard model"""
-        self.logTestName()
-        self.eng.reset()
-        q = self.eng.register
+        prog = sf.Program(2)
         H = bose_hubbard(1, 2, self.J, 0)
 
-        with self.eng:
+        with prog.context as q:
             Fock(2) | q[0]
             BoseHubbardPropagation(H, self.t, self.k) | q
 
-        state = self.eng.run('fock', cutoff_dim=7)
+        state = eng.run(prog).state
 
         Hm = -self.J*np.sqrt(2)*np.array([[0, 1, 0], [1, 0, 1], [0, 1, 0]])
         init_state = np.array([1, 0, 0])
         exp = np.abs(np.dot(expm(-1j*self.t*Hm), init_state))**2
 
-        self.assertTrue(np.allclose(state.fock_prob([2, 0]), exp[0], rtol=self.tol))
-        self.assertTrue(np.allclose(state.fock_prob([1, 1]), exp[1], rtol=self.tol))
-        self.assertTrue(np.allclose(state.fock_prob([0, 2]), exp[2], rtol=self.tol))
+        assert np.allclose(state.fock_prob([2, 0]), exp[0], rtol=tol)
+        assert np.allclose(state.fock_prob([1, 1]), exp[1], rtol=tol)
+        assert np.allclose(state.fock_prob([0, 2]), exp[2], rtol=tol)
 
-    def test_1x2(self):
+    def test_1x2(self, eng, tol):
         """Test a 1x2 lattice Bose-Hubbard model"""
-        self.logTestName()
-        self.eng.reset()
-        q = self.eng.register
+        prog = sf.Program(2)
         H = bose_hubbard(1, 2, self.J, self.U)
 
-        with self.eng:
+        with prog.context as q:
+            Fock(2) | q[0]
+            BoseHubbardPropagation(H, self.t, self.k) | q
+
+        state = eng.run(prog).state
+
+        Hm = -self.J*np.sqrt(2)*np.array([[0, 1, 0], [1, 0, 1], [0, 1, 0]]) \
+            + self.U*np.diag([1, 0, 1])
+        init_state = np.array([1, 0, 0])
+        exp = np.abs(np.dot(expm(-1j*self.t*Hm), init_state))**2
+
+        assert np.allclose(state.fock_prob([2, 0]), exp[0], rtol=tol)
+        assert np.allclose(state.fock_prob([1, 1]), exp[1], rtol=tol)
+        assert np.allclose(state.fock_prob([0, 2]), exp[2], rtol=tol)
+
+    def test_1x2_quadrature_operators(self, eng, tol):
```

Comment:
new test?

Reply:
Yep, noticed it wasn't being tested previously, and GitHub/Codecov wouldn't pass the coverage checks without it